tests/test_pf7.py

```diff
@@ -0,0 +1,433 @@
+            pf7.extended_calldata_variables, self.test_extended_calldata_variables
+        )
+        self.assertEqual(pf7.extended_variant_fields, self.test_extended_variant_fields)
+        mock_load_config.assert_called_once_with(None)
+        mock_init_filesystem.assert_called_once_with(self.url_starts_with_gs)
+
+    @patch("malariagen_data.pf7.init_filesystem", return_value=["fs", "path"])
+    def test_setup_with_config_set(self, mock_init_filesystem):
+        with patch(
+            "malariagen_data.pf7.Pf7._load_config",
+            return_value=self.config_content,
+        ) as mock_load_config:
+            pf7 = Pf7(self.url_starts_with_gs, data_config=self.test_config_path)
+        self.assertEqual(pf7._cache_sample_metadata, None)
+        self.assertEqual(pf7._cache_zarr, None)
+        self.assertEqual(
+            pf7.extended_calldata_variables, self.test_extended_calldata_variables
+        )
+        self.assertEqual(pf7.extended_variant_fields, self.test_extended_variant_fields)
+        mock_load_config.assert_called_once_with(self.test_config_path)
+        mock_init_filesystem.assert_called_once_with(self.url_starts_with_gs)
+
+    def test_load_config(self):
+        config = self.test_pf7_class._load_config(self.test_config_path)
+        self.assertEqual(
+            config,
+            {
+                "metadata_path": "metadata/test_metadata.txt",
+                "zarr_path": "pf7.zarr/",
+            },
+        )
+
+    def test_sample_metadata_returns_extpected_df(self):
+        metadata_df = self.test_pf7_class.sample_metadata()
+        pd.testing.assert_frame_equal(metadata_df, self.test_metadata_df)
+
+    def test_read_general_metadata_returns_expected_df_with_cache_set(
+        self,
+    ):
+        self.test_pf7_class.sample_metadata()
+        metadata_df = self.test_pf7_class.sample_metadata()
+        pd.testing.assert_frame_equal(metadata_df, self.test_metadata_df)
+
+    @patch("builtins.open", new_callable=mock_open)
+    @patch("malariagen_data.pf7.pd.read_csv", return_value=pd.DataFrame())
+    def test_read_general_metadata_uses_cache_when_set(self, mock_read_csv, mock_open):
+        self.test_pf7_class.sample_metadata()
+        self.test_pf7_class.sample_metadata()
+        mock_open.assert_called_once()
+        mock_read_csv.assert_called_once()
+
+    @patch("malariagen_data.pf7.init_zarr_store", return_value="Safe store object")
+    @patch("malariagen_data.pf7.zarr.open_consolidated")
+    def test_open_zarr_calls_functions_correctly(self, mock_zarr, mock_safestore):
+        with patch(
+            "malariagen_data.pf7.init_filesystem",
+            return_value=["fs", self.test_data_path],
+        ):
+            pf7_mock_fs = Pf7(self.test_data_path, data_config=self.test_config_path)
+        pf7_mock_fs.open_zarr()
+        mock_safestore.assert_called_once_with(fs="fs", path=self.test_zarr_path)
+        mock_zarr.assert_called_once_with(store="Safe store object")
```

Comment:
Excess baggage?

Reply:
I have removed this as the same thing is basically tested at the same time as testing the cache